hist/hist/inc/TH1.h

```diff
@@ -630,7 +630,11 @@ class TH1F : public TH1, public TArrayF {
    TH1F& operator=(const TH1F &h1);
    ~TH1F() override;
 
-   void     AddBinContent(Int_t bin) override {++fArray[bin];}
+   /// Increment bin content by 1.
+   /// Passing an out-of-range bin leads to undefined behavior
+   void     AddBinContent(Int_t bin) override {if (fArray[bin] < 16777216.f) ++fArray[bin];}
+   /// Increment bin content by a weight Float_t(w).
+   /// Passing an out-of-range bin leads to undefined behavior
    void     AddBinContent(Int_t bin, Double_t w) override
                           { fArray[bin] += Float_t (w); }
```

Comment:
I would add the same check also in case of weights. In this case it needs to be computed at run time :
```
float bin_content_max = w/(0.5 * std::numeric_limits<float>::epsilon());
```

Reply:
I think this check does not make sense, because I might call AddBinContent with differents weight each time within the same hist. So it would mess up with the limit. bin_content_max would change dynamically within the same histogram.

For example:

AddBinContent(1, 10);
AddBinContent(2, 0.1);

The limit for bin 1 would be different than for bin 2.

So this does not get towards a fixed limit...